Sources/ArcGISToolkit/Components/Form/FormInputs/DateTimeInput.swift

```diff
@@ -61,23 +67,32 @@ struct DateTimeInput: View {
         }
         .padding([.bottom], elementPadding)
         .onAppear {
-            if element.value.isEmpty {
+            if inputModel.value.isEmpty {
                 date = nil
             } else {
-                date = try? Date(element.value, strategy: .arcGISDateParseStrategy)
+                date = try? Date(inputModel.value, strategy: .arcGISDateParseStrategy)
             }
         }
         .onChange(of: date) { newDate in
-            guard let currentDate = try? Date(element.value, strategy: .arcGISDateParseStrategy),
+            guard let currentDate = try? Date(inputModel.value, strategy: .arcGISDateParseStrategy),
                   newDate != currentDate else {
                 return
             }
-            requiredValueMissing = element.isRequired && newDate == nil
+            requiredValueMissing = inputModel.isRequired && newDate == nil
             featureForm?.feature.setAttributeValue(newDate, forKey: element.fieldName)
+            model.evaluateExpressions()
         }
         .onChange(of: model.focusedFieldName) { newFocusedFieldName in
             isEditing = newFocusedFieldName == element.fieldName
         }
+        .onChange(of: inputModel.value) { newValue in
+            if newValue.isEmpty {
+                date = nil
+            } else {
+                date = try? Date(newValue, strategy: .arcGISDateParseStrategy)
+            }
+            // TODO: make sure this triggers `.onChange(of: date)`
```

Comment:
I tried this TODO out for you. Seems like we're good to go and it can be removed.

Using the **Date Expressions** form (`ed7b8fffd9bc408081a847eeda283123`)

![Simulator Screen Recording - iPhone 15 Pro Max - 2023-10-10 at 16 17 43](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/06d68dcd-c9f5-4e75-99c5-1536583ea6f5)

```suggestion
```